app/src/main/java/org/mozilla/fenix/settings/address/view/AddressEditorView.kt

```diff
@@ -103,4 +147,32 @@ class AddressEditorView(
             create()
         }.show()
     }
+
+    private fun bindCountryDropdown() {
+        val adapter = ArrayAdapter<String>(
+            binding.root.context,
+            android.R.layout.simple_spinner_dropdown_item,
+        )
+
+        var selectedPosition = -1
+        var regionPosition = -1
+        var defaultPosition = -1
+        availableCountries.values.forEachIndexed { index, country ->
+            if (country.key == defaultCountry.key) defaultPosition = index
+            if (country.key == region?.home) regionPosition = index
+            if (country.key == address?.country) selectedPosition = index
+            adapter.add(country.displayName)
+        }
+
+        val setPosition = selectedPosition.takeIf { it != -1 }
+            ?: regionPosition.takeIf { it != -1 }
+            ?: defaultPosition.takeIf { it != -1 }
+            ?: 0
+        binding.countryDropDown.adapter = adapter
+        binding.countryDropDown.setSelection(setPosition)
+    }
+
+    private fun String.toCountryCode() = availableCountries.values.find {
+        it.displayName == this
+    }?.key ?: defaultCountry.key
```

Comment:
`?: defaultCountry.key` looks particularly scary to me since you would hope we would never hit such a case, but knowing that we shouldn't ever hit this, it makes me wonder if we should even have it.

Reply:
On further thought, I am reasoning that this is fine since `find {}` will return something that could be nullable.